File: Cpp/Majority Number.cpp
class Solution {
public:
    /**
     * @param nums: A list of integers
     * @return: The majority number
     */
    int majorityNumber(vector<int> nums) {
        // write your code here
        int cnt = 1, res = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (cnt == 0) {
                res = nums[i];
                cnt = 1;
            }
            if (nums[i] == res) {
                cnt ++;
            } else {
                cnt --;
            }
        }
        return res;
    }
};

File: Python/Majority-Number.py
class Solution:
    """
    @param nums: A list of integers
    @return: The majority number
    """
    def majorityNumber(self, nums):
        # write your code here
        cur = -1
        times = 0
        for i in nums:
            if times == 0:
                times = 1
                cur = i
            elif i == cur:
                times += 1
            elif i != cur:
                times -= 1
        return cur